presto-orc/src/main/java/com/facebook/presto/orc/reader/ListSelectiveStreamReader.java

```diff
@@ -400,17 +418,24 @@ else if (listFilter != null && listFilter.getChild() != null) {
         return streamPosition;
     }
 
-    private void populateElementPositions(int elementPositionCount)
+    private int populateElementPositions(int elementPositionCount)
     {
         elementPositions = ensureCapacity(elementPositions, elementPositionCount);
 
         int index = 0;
         for (int i = 0; i < outputPositionCount; i++) {
-            for (int j = 0; j < elementLengths[i]; j++) {
+            int length = elementLengths[i];
```

Comment:
we can eliminate the length variable and assign elementLengths[i] = maxElementLength

Reply:
Are you asking for the following? Seems less readable.

```
            if (maxElementLength != ELEMENT_LENGTH_UNBOUNDED && elementLengths[i] > maxElementLength) {
                elementLengths[i] = maxElementLength;
            }

            for (int j = 0; j < elementLengths[i]; j++) {
                elementPositions[index] = elementOffsets[i] + j;
                index++;
            }
```

Reply:
We can do it without condition.
elementLengths[i] = Math.max(elementLength[i], maxElementLength)

Reply:
@swapsmagic I don't think this works.

- maxElementLength = 10, elementLength[i] = 5 -> 10 - wrong, should be 5
- maxElementLength = 10, elementLength[i] = 15 -> 15- wrong, should be 10

Reply:
sorry my bad.